### PR TITLE
[AMD] fix(triton-mla): cap max_kv_splits at 256 on gfx942 (Kimi-K2.6 hang)

### PR DESCRIPTION
**Pull request by bingxche**
## Motivation

Fix the `nightly-8-gpu-kimi-k26` MI325X hang reproduced in https://github.com/sgl-project/sglang/actions/runs/25513282022/job/74877480809 (scheduler watchdog at 300 s, MLA decode kernel page-faulting with `Memory access fault by GPU node ... Reason: Write access to a read-only page`).

Bisect window `d86f2916..7d397ad2` → #20479 / `3fe8bc987`. Its `_mla_decode_kv_splits_cap()` lifts `max_kv_splits` to `next_power_of_2(sm_count)`. On gfx942 (MI300X / MI325X, `sm_count = 304`) this rounds up to **512** — twice gfx950 (MI355X)'s natural value of 256. The Triton-MLA persistent `cuda_graph_attn_logits` buffer

```
(cuda_graph_max_bs, num_head, max_kv_splits, v_head_dim) float32
```

then balloons to `512 × 8 × 512 × 512 × 4 B = 4 GiB` on Kimi-K2.6 + TP=8 (vs ~2 GiB on gfx950 with the same model), and ROCm faults during CUDA graph replay. The same workflow run's `nightly-8-gpu-mi35x-kimi-k26` job already passes end-to-end on gfx950 at the 256-splits / 2 GiB operating point, so 256 is empirically known to be safe for this model.

## Modifications

Three files, +25 / -7, NVIDIA path untouched.

`python/sglang/srt/utils/common.py`: new `is_gfx942_supported()` utility, symmetric to the existing `is_gfx95_supported()`.

`python/sglang/srt/layers/attention/triton_backend.py`: clamp `max_kv_splits` to 256 on gfx942 only.

```python
if _is_gfx942:
    # gfx942 (MI300X / MI325X) has 304 CUs, so #20479's next_power_of_2(sm_count)
    # rounds up to 512 — twice MI355X's natural cap of 256 — and the persistent
    # cuda_graph_attn_logits fp32 buffer hits ~4 GiB on Kimi-K2.6 (v_head_dim=512),
    # faulting in ROCm CUDA graph replay
    # (https://github.com/sgl-project/sglang/actions/runs/25513282022/job/74877480809).
    # Pin the cap at 256 so gfx942 matches the gfx950 (MI355X) behavior that we
    # already validated end-to-end.
    self.max_kv_splits = min(self.max_kv_splits, 256)
```

`test/registered/amd/test_kimi_k2_instruct.py`: cap GSM8K `parallel` from 1319 to 512. Once the page-fault is fixed (above), the persistent `cuda_graph_attn_logits` buffer still occupies ~2 GiB on gfx942 vs the pre-#20479 ~128 MiB, leaving less inference-time headroom for the high-concurrency prefill workload than the original test was designed against on MI300X (192 GB HBM). Capping `parallel` at `cuda_graph_max_bs` keeps decode batches inside the pre-captured CUDA graph range and avoids eager-mode dynamic allocation that would otherwise OOM. All 1319 questions are still evaluated; only client-side concurrency is bounded.

| Setup | #20479 `sm_cap` | this fix | final `max_kv_splits` | `cuda_graph_attn_logits` |
|---|---|---|---|---|
| **MI325X K2.6 TP=8** | 512 | **256** | **256** | **2 GiB** ✅ |
| MI355X K2.6 TP=8 (gfx950, untouched) | 256 | — | 256 | 2 GiB ✅ |
| H100 K2.6 (NVIDIA, untouched) | 256 | — | 256 | 2 GiB ✅ |

## Validation
Test passed 
https://github.com/sgl-project/sglang/actions/runs/25847861313/job/75956509768
image

https://github.com/sgl-project/sglang/actions/runs/25787988241/job/75746077178
image

## Checklist

- [x] Format your code according to the Format code with pre-commit.
- [x] Add unit tests according to the Run and add unit tests. (Validated end-to-end on MI325X via the failing nightly suite; accuracy 0.940.)
- [ ] Update documentation according to Write documentations.
- [x] Provide accuracy and speed benchmark results.
- [x] Follow the SGLang code style guidance.

<div><a href="https://cursor.com/agents/bc-ea90a4e1-da18-4f64-816f-e74904de5f29"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-ea90a4e1-da18-4f64-816f-e74904de5f29"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

---
### CI States

Latest PR Test (Base): :x: [Run #26731869912](https://github.com/sgl-project/sglang/actions/runs/26731869912)
Latest PR Test (Extra): :x: [Run #26731869849](https://github.com/sgl-project/sglang/actions/runs/26731869849)

### CODE REVIEW

**Review by HaiShaw — APPROVED** [review]

**Review by HaiShaw — CHANGES_REQUESTED** [review]
TBD

File: python/sglang/srt/layers/attention/triton_backend.py [hunk 1]
```diff
@@ -153,6 +155,13 @@ def __init__(
                 self.device_core_count,
                 self.max_context_len,
             )
+            if _is_hip:
+                # Bound attn_logits float32 buffer to 512 MB on ROCm; #20479's sm_cap faults at ~2 GB on MI3xx + Kimi-K2.6 (https://github.com/sgl-project/sglang/actions/runs/25513282022/job/74877480809).
+                bs = model_runner.server_args.cuda_graph_max_bs or 1
```
Comment (HaiShaw) [inline python/sglang/srt/layers/attention/triton_backend.py]: please use different var name than `bs`, be more specific.

**Review by HaiShaw — CHANGES_REQUESTED** [review]

**Review by kkHuang-amd — APPROVED** [review]
LGTM

**Review by HaiShaw — APPROVED** [review]

**Comment by bingxche** [comment]
@amd-bot ci-status
